das/daser.go

```diff
@@ -90,5 +148,97 @@ func (d *DASer) sampling(ctx context.Context, sub header.Subscription) {
 		sampleTime := time.Since(startTime)
 		log.Infow("sampling successful", "height", h.Height, "hash", h.Hash(),
 			"square width", len(h.DAH.RowsRoots), "finished (s)", sampleTime.Seconds())
+
+		checkpoint = h.Height
+	}
+}
+
+// catchUpJob represents a catch-up job. (from:to]
+type catchUpJob struct {
+	from, to int64
+}
+
+// catchUpManager manages catch-up jobs, performing them one at a time, exiting
+// only once context is canceled and storing latest DASed checkpoint to disk.
+func (d *DASer) catchUpManager(ctx context.Context, checkpoint int64) {
+	defer func() {
+		// store latest DASed checkpoint to disk here to ensure that if DASer is not yet
+		// fully caught up to network head, it will resume DASing from this checkpoint
+		// up to current network head
+		// TODO @renaynay: Implement Share Cache #180 to ensure no duplicate DASing over same
+		//  header
+		if err := storeCheckpoint(d.cstore, checkpoint); err != nil {
+			log.Errorw("storing checkpoint to disk", "height", checkpoint, "err", err)
+		}
+		// signal that all catchUp routines have finished
```

Comment:
doc nit: we now how only one catchUp routine

Reply:
resolved? https://github.com/celestiaorg/celestia-node/pull/473/commits/ad2975d10455ed92488bdbd81d5c30cf59e3a0bf